x/nodes/keeper/slash.go

```diff
@@ -8,12 +8,34 @@ import (
 	"github.com/pokt-network/pocket-core/x/nodes/types"
 	"github.com/tendermint/tendermint/crypto"
 
+	"github.com/pokt-network/pocket-core/codec"
 	sdk "github.com/pokt-network/pocket-core/types"
 )
 
 // BurnForChallenge - Tries to remove coins from account & supply for a challenged validator
 func (k Keeper) BurnForChallenge(ctx sdk.Ctx, challenges sdk.BigInt, address sdk.Address) {
-	coins := k.RelaysToTokensMultiplier(ctx).Mul(challenges)
+	var coins sdk.BigInt
+	//check if PIP22 is enabled, if so scale the rewards
+	if k.Cdc.IsAfterNamedFeatureActivationHeight(ctx.BlockHeight(), codec.RSCALKey) {
+		//grab stake
+		validator, found := k.GetValidator(ctx, address)
+		if !found {
+			ctx.Logger().Error(fmt.Errorf("no validator found for address %s; at height %d\n", address.String(), ctx.BlockHeight()).Error())
+			return
+		}
+
+		stake := validator.GetTokens()
+		//floorstake to the lowest bin multiple or take ceiling, whicherver is smaller
+		flooredStake := sdk.MinInt(stake.Sub(stake.Mod(k.ServicerStakeFloorMultiplier(ctx))), (k.ServicerStakeWeightCeiling(ctx).Sub(stake.Mod(k.ServicerStakeFloorMultiplier(ctx)))))
+		//Convert from tokens to a BIN number
+		Bin := flooredStake.Quo(k.ServicerStakeFloorMultiplier(ctx))
+		//calculate the weight value
+		weight := Bin.ToDec().FracPow(k.ServicerStakeFloorMultiplierExponent(ctx), 100).Quo(k.ServicerStakeWeightMultiplier(ctx))
```

Comment:
See comment about hardcoded value here: `x/nodes/keeper/reward.go`.

Reply:
addressed in [7f6fbed](https://github.com/pokt-network/pocket-core/pull/1449/commits/7f6fbed326cae72229041a71f1735055a82f425f)